packages/gatsby/src/bootstrap/load-plugins/__tests__/load-plugins.ts

```diff
@@ -187,4 +199,105 @@ describe(`Load plugins`, () => {
       expect(tsplugins.length).toEqual(2)
     })
   })
+
+  describe(`plugin options validation`, () => {
+    it(`throws a structured error with invalid plugin options`, async () => {
+      const invalidPlugins = [
+        {
+          resolve: `gatsby-plugin-google-analytics`,
+          options: {
+            trackingId: 123,
+            anonymize: `not a boolean`,
+          },
+        },
+        {
+          resolve: `gatsby-plugin-google-analytics`,
+          options: {
+            anonymize: `still not a boolean`,
+          },
+        },
+      ]
+      await loadPlugins({
+        plugins: invalidPlugins,
+      })
+
+      expect(reporter.error as jest.Mock).toHaveBeenCalledTimes(
+        invalidPlugins.length
+      )
+      expect((reporter.error as jest.Mock).mock.calls[0])
+        .toMatchInlineSnapshot(`
+        Array [
+          Object {
+            "context": Object {
+              "pluginName": "gatsby-plugin-google-analytics",
+              "validationErrors": Array [
+                Object {
+                  "context": Object {
+                    "key": "trackingId",
+                    "label": "trackingId",
+                    "value": 123,
+                  },
+                  "message": "\\"trackingId\\" must be a string",
+                  "path": Array [
+                    "trackingId",
+                  ],
+                  "type": "string.base",
+                },
+                Object {
+                  "context": Object {
+                    "key": "anonymize",
+                    "label": "anonymize",
+                    "value": "not a boolean",
+                  },
+                  "message": "\\"anonymize\\" must be a boolean",
+                  "path": Array [
+                    "anonymize",
+                  ],
+                  "type": "boolean.base",
+                },
+              ],
+            },
+            "id": "11331",
+          },
+        ]
+      `)
```

Comment:
Do we want inlineSnapshot?

Reply:
Personal preference — I like inline snapshot a lot more because with `toMatchSnapshot` the result is hidden in some folder and file that nobody ever looks at / can understand! If you prefer to move it out, I'd be happy to use `.tomatchSnapshot` though!

Reply:
Honestly, I don't care but we usually use matchSnapshot, that's why I asked